src/modules/usart.rs

```diff
@@ -0,0 +1,258 @@
+        })
+    }
+
+    /// Returns `true` if a `char` is available to read.
+    pub fn readable(&self) -> bool {
+        (self.usart.IF & usart::STATUS_RXDATAV) != 0
+    }
+
+    /// Returns `true` if there is space available to write a `char`.
+    pub fn writeable(&self) -> bool {
+        (self.usart.IF & usart::STATUS_TXBL) != 0
+    }
+
+
+    fn init_location_and_gpio(&mut self) {
+        let location = match self.config {
+            Config::Usart0(loc) => {
+                match loc {
+                    Loc0 => {
+                        self.init_gpio(Some((Port::E, 10)), Some((Port::E, 11)),
+                            Some((Port::E, 12)), Some((Port::E, 13)));
+                        usart::ROUTE_LOCATION_LOC0
+                    },
+                    Loc1 => {
+                        self.init_gpio(Some((Port::E, 7)), Some((Port::E, 6)),
+                            Some((Port::E, 5)), Some((Port::E, 4)));
+                        usart::ROUTE_LOCATION_LOC1
+                    },
+                    Loc2 => {
+                        self.init_gpio(Some((Port::C, 11)), Some((Port::C, 10)),
+                            Some((Port::C, 9)), Some((Port::C, 8)));
+                        usart::ROUTE_LOCATION_LOC2
+                    },
+                    Loc3 => {
+                        self.init_gpio(Some((Port::E, 13)), Some((Port::E, 12)),
+                            None, None);
+                        usart::ROUTE_LOCATION_LOC3
+                    },
+                    Loc4 => {
+                        self.init_gpio(Some((Port::B, 7)), Some((Port::B, 8)),
+                            Some((Port::B, 13)), Some((Port::B, 14)));
+                        usart::ROUTE_LOCATION_LOC4
+                    },
+                    Loc5 => {
+                        self.init_gpio(Some((Port::C, 0)), Some((Port::C, 1)),
+                            Some((Port::B, 13)), Some((Port::B, 14)));
+                        usart::ROUTE_LOCATION_LOC5
+                    },
+                }
+            },
+            Config::Usart1(loc) => {
+                match loc {
+                    Loc0 => {
+                        self.init_gpio(Some((Port::C, 0)), Some((Port::C, 1)),
+                            Some((Port::B, 7)), Some((Port::B, 8)));
+                        usart::ROUTE_LOCATION_LOC0
+                    },
+                    Loc1 => {
+                        self.init_gpio(Some((Port::D, 0)), Some((Port::D, 1)),
+                            Some((Port::D, 2)), Some((Port::D, 3)));
+                        usart::ROUTE_LOCATION_LOC1
+                    },
+                    Loc2 => {
+                        self.init_gpio(Some((Port::D, 7)), Some((Port::D, 6)),
+                            Some((Port::F, 0)), Some((Port::F, 1)));
+                        usart::ROUTE_LOCATION_LOC2
+                    },
+                    _ => panic!("Invalid location for Usart1: {:?}", loc)
```

Comment:
Maybe, if so I don't know how. An error such as this would show up very quickly either way, as one probably wouldn't just try to instantiate any Usarts without knowing what pins to use.
It's also possible to return a Result<(), UsartError> or something like that, and make it up to the user of how to handle it.